2018/DVE-2018-0002.md

```diff
@@ -0,0 +1,100 @@
+# DVE-2018-0002: Microsoft server replies FORMERR for unknown EDNS options
+
+## Description
+
+86.64.239.11 is one of the authoritative name servers for sfr.net (SFR being a big ISP in France). It handles EDNS except when the client uses an option it doesn't know, like NSID or cookies. It then replies FORMERR.
+
+It is clearly a violation of RFC 6891 "Any OPTION-CODE values not understood by a responder or requestor MUST be ignored."
+
+## Evidence
+
+```
+% dig +dnssec +bufsize=4096 @86.64.239.11 NS sfr.net
+
+; <<>> DiG 9.10.3-P4-Debian <<>> +dnssec +bufsize=4096 @86.64.239.11 NS sfr.net
+; (1 server found)
+;; global options: +cmd
+;; Got answer:
+;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 27995
+;; flags: qr aa rd; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 3
+;; WARNING: recursion requested but not available
```

Comment:
Use `+norec` when querying an authoritative nameserver. It doesn't make a difference in most(?) cases other than getting rid of this warning message, but in theory a query could be proxied to a different backend server based on the RD bit.